providers/implementations/rands/seeding/rand_vms.c

```diff
@@ -38,6 +38,32 @@
 # error "Unsupported seeding method configured; must be os"
 #endif
 
+#define PUBLIC_VECTORS "SYS$LIBRARY:SYS$PUBLIC_VECTORS.EXE"
+
+/*
+ * Helper functions
+ * ----------------
+ *
+ * gettime() gets time using sys$gettim_prec or if that's not available,
+ * sys$gettim.  sys$gettim_prec exists since OpenVMS v8.4.
+ */
+#define GETTIM_PREC "SYS$GETTIM_PREC"
+
+static int gettime(unsigned __int64 *t)
+{
+    static int gettim_prec_address_flag = 0;
+    static int (*gettim_prec_address)(unsigned __int64 *t) = NULL;
+
+    if (!gettim_prec_address_flag) {
+        gettim_prec_address = dlsym(dlopen(PUBLIC_VECTORS, 0), GETTIM_PREC);
```

Comment:
There is a potential race here. If function pointer assignment is atomic then its probably not a big problem. Otherwise a lock might be required. At the very least a comment might be appropriate pointing out the race and that we accept it.

Reply:
Does VMS offer pthread_once?